src/mlpack/methods/ann/brnn_impl.hpp

```diff
@@ -60,6 +60,22 @@ BRNN<OutputLayerType, MergeLayerType, MergeOutputType,
   /* Nothing to do here. */
 }
 
+template<typename OutputLayerType, typename MergeLayerType,
+         typename MergeOutputType, typename InitializationRuleType,
+         typename... CustomLayers>
+BRNN<OutputLayerType, MergeLayerType, MergeOutputType,
+    InitializationRuleType, CustomLayers...>::~BRNN()
+{
+  // Remove mergeLayer from the forward and backward RNNs so it doesn't get
+  // deleted.  This assumes that mergeLayer is the last layer!
+  forwardRNN.network.pop_back();
+  backwardRNN.network.pop_back();
```

Comment:
This doesn't look right. We never add the mergeLayer to the forward and backward RNN but the last layer of the RNN's to the merge layer if you look [here](https://github.com/mlpack/mlpack/blob/3199df34fef4eb10ba26487561d7d678b04e6fe9/src/mlpack/methods/ann/brnn_impl.hpp#L635-L642).
I guess we only need change the comment though. If reset is true then we should pop the last layers from the RNN's to avoid deleting twice otherwise we should simply delete the mergeLayer

Reply:
Oh, you're right!  I read it backwards.  In any case, yeah, removing the last layer from the forward and backward RNNs is sufficient to prevent a memory leak.  I've updated the comment in c42b9f767. :+1: